Shopify/tophat
# Clean up naming and other missed changes

### What does this change accomplish?

This change circles back to some names and missed changes to clean them up.  Namely:

### How have you achieved it?

- Entities previously referred to as "sources" are now consistently called "recipes."
- `RemoteArtifactSource` is now just `ArtifactSource`.
- Cleaned up all old missed references to "builds" to instead be called "artifacts."
- Updated SwiftData configuration to write to a Tophat-specific file.
- Minified the icon updating task into the `InstallSession` as it was previously overly-complex.

### How can the change be tested?

There should be no change in functionality.  Ensure that Tophat generally works as expected.
